contracts/modules/Module.sol

```diff
@@ -81,12 +85,25 @@ contract Module is IModule, ModuleStorage, Pausable {
         require(token.transfer(msg.sender, balance), "Transfer failed");
     }
 
-   /**
-    * @notice Reclaims ETH
-    * @dev We duplicate here due to the overriden owner & onlyOwner
-    */
+    /**
+     * @notice Reclaims ETH
+     * @dev We duplicate here due to the overriden owner & onlyOwner
+     */
     function reclaimETH() external {
         _onlySecurityTokenOwner();
         msg.sender.transfer(address(this).balance);
     }
+
+    /**
+     * @notice Deduct the usage cost
+     */
+    function _deductUsageFee() internal {
+        uint256 _usageCost = IModuleFactory(factory).usageCostInPoly();
```

Comment:
This means that the factory owner can arbitrarily change usage cost. Front running attacks are also possible where the issuer initiates the transaction thinking that fee is X but module owner front runs that tx to increase the usage fee to Y.

Reply:
The issuer can reduce the risk by properly using the budgeting in the security token but I think we can come up with a better solution.

Reply:
lockup period is used to avoid front running issues. Factory owner can change the value of usage cost in 24 hours only because of this front running chances are very low.

Reply:
In its current implementation, the lock period is failing to do that. It's just limiting the module owner to do one front-run attack per 24 hours. Ideally, Even one attack shouldn't be possible.

If we implement something like what I mentioned in the other review comment, both this and that review comment will be solved.

Reply:
hmm, Both cases are probabilistic but anyway I don't have any problem to implement.